src/extension/src/coreTemplateStudio.ts

```diff
@@ -49,40 +47,32 @@ export class CoreTemplateStudio {
       extensionPath = path.join(__dirname, "..");
     }
 
-    if (platform === CONSTANTS.API.WINDOWS_PLATFORM_VERSION) {
-      executableName += ".exe";
+    if (platform === CONSTANTS.CLI.WINDOWS_PLATFORM_VERSION) {
+      cliExecutableName += ".exe";
     }
 
-    let apiPath = path.join(
+    let cliPath = path.join(
       extensionPath,
       "src",
-      "api",
+      "corets-cli",
       platform,
-      executableName
+      cliExecutableName
     );
 
-    let apiWorkingDirectory = path.join(extensionPath, "src", "api", platform);
+    let cliWorkingDirectory = path.join(extensionPath, "src", "corets-cli", platform);
 
-    if (os.platform() !== CONSTANTS.API.WINDOWS_PLATFORM_VERSION) {
+    if (os.platform() !== CONSTANTS.CLI.WINDOWS_PLATFORM_VERSION) {
       // Not unsafe as the parameter comes from trusted source
       // tslint:disable-next-line:non-literal-fs-path
-      fs.chmodSync(apiPath, 0o755);
+      fs.chmodSync(cliPath, 0o755);
     }
 
-    const port = await portfinder.getPortPromise({
-      port: CONSTANTS.START_PORT
+    const spawnedProcessCli = spawn(cliPath, [], {
```

Comment:
Curious does this work on mac, with the cwd? I remember I had problems with getting spawn to keep the cwd on mac. Which caused the relative paths passed in to not work. Can you just make sure of this (Try both local dev host and vsix). Might have just been an API thing though, or node fixed the problem.

Reply:
I really hope it works because we need the spawn shell to redirect stdio

Reply:
If no one has time to verify this right now, I can check this in ~48 hours (exams almost over 😄)

Reply:
In the original API call, they also use cwd for all platforms. We tested in Mac, it ran into the same error message Jason got in his Windows. Investigating the issue right now.

Reply:
Cool! just wanted to make sure since I remember having to use exec, guess its been fixed! (Or that it was an older mac version problem).